pulp_2_tests/tests/rpm/cli/test_sync.py

```diff
@@ -202,6 +207,44 @@ def _list_units(cfg, unit_type):
         )).stdout.splitlines()
 
 
+class ForceSyncDuplicateSRPMFailure(_BaseTestCase):
+    """Test a forced full sync with --srpm skip does not fail.
+
+    This test case targets `Pulp #4397`_ and `Pulp #4459`_.
+    The test procedure is as follows:
+
+    1. Create and sync a repository containing duplicate SRPMs with --skip
+    2. Sync the repository with ``--force-full``. Verify sync is successful.
+
+    This test case has added distribution units and could be expanded to other unit types.
+
+    .. _Pulp #4397: https://pulp.plan.io/issues/4397
+    .. _Pulp #4459: https://pulp.plan.io/issues/4459
+    """
+
+    def test_all(self):
+        """Test whether Pulp can synch with --srpm skip."""
+        cfg = config.get_config()
+        if cfg.pulp_version < Version('2.19'):
+            raise unittest.SkipTest('This test requires Pulp 2.19 or newer.')
+        repo_id = utils.uuid4()
+        client = cli.Client(cfg)
+        client.run((
+            'pulp-admin', 'rpm', 'repo', 'create', '--repo-id', repo_id,
+            '--feed', SRPM_DUPLICATE_FEED_URL, '--skip', 'srpm',
+        ))
+        self.addCleanup(client.run, (
+            'pulp-admin', 'rpm', 'repo', 'delete', '--repo-id', repo_id,
+        ))
+        # Verify synch does not contain BZ error
+        proc = sync_repo(cfg, repo_id)
```

Comment:
Missing second step.

```suggestion
  # Sync the repository with --force-full.
  proc = sync_repo(self.cfg, repo_id, force_sync=True)
```

Reply:
Perhaps, we will need to re-evaluate those assertions, I ran this test with `force_sync=True`, and the test passed.